src/types.ts

```diff
@@ -0,0 +1,148 @@
+import Client from "./client";
+import Context from "./context";
+import { ContextDataProvider } from "./provider";
+import { ContextPublisher } from "./publisher";
+// eslint-disable-next-line no-shadow
+import { AbortSignal } from "./abort-controller-shim";
+
+export type AbortControllerEvents = {
+	[key: string]: Array<() => unknown>;
+};
+
+export type Exposure = {
+	id: number;
+	name: string;
+	exposedAt: number;
+	unit: string | null;
+	variant: number;
+	assigned: boolean;
+	eligible: boolean;
+	overridden: boolean;
+	fullOn: boolean;
+	custom: boolean;
+	audienceMismatch: boolean;
+};
+
+export type Attribute = {
+	name: string;
+	value: unknown;
+	setAt: number;
+};
+
+export type Units = {
+	[key: string]: string | number;
+};
+
+export type Goal = {
+	name: string;
+	properties: Record<string, unknown> | null;
+	achievedAt: number;
+};
+
+export type EventLogger = (context: Context, eventName: string, data?: Record<string, unknown> | Error) => void;
```

Comment:
I think we could be more specific here for `data` - it could be an Exposure, Goal, PublishEvent, Error, and perhaps something else for the ready event?

Reply:
Done :)